scripts/drs_file_access_examination.ipynb

```diff
@@ -0,0 +1,326 @@
+    "\n",
+    "\n",
+    "The Data Repository Service (DRS) API is a standardized set of access methods that are agnostic to cloud infrastructure. Developed by the Global Alliance for Genomics and Health (GA4GH), DRS enable researchers to access data regardless of the underlying architecture of the repository (i.e. Google Cloud, Azure, AWS, etc.) in which it is stored. Terra supports accessing data using the GA4GH standard Data Repository Service (DRS). To learn more look at this link: https://support.terra.bio/hc/en-us/articles/360039330211-Data-Access-with-the-GA4GH-Data-Repository-Service-DRS- \n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "To run code from the terra-notebook-utils DRS library, we first have to install the package and restart the kernel. \n",
+    "\n",
+    "1. Run the cell below\n",
+    "2. Go to the kernel tab in the menu at the top, and select restart\n",
+    "3. Press the restart in the pop-up menu"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Installing terra-notebook-utils library\n",
+    "!pip install terra-notebook-utils"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Grab all DRS urls from the data table\n",
+    "\n",
+    "The DRS url is stored under the object_id column in a table.\n",
+    "\n",
+    "Using the firecloud library, we are going to search for the object_id column in each table in the data table. If a table contains an object_id column, we check if the content is a string and has 'drs://' at its beginning."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "code_folding": [],
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from firecloud import api as fapi\n",
+    "\n",
+    "# Getting the namespace/project and name of the workspace\n",
+    "namespace = os.environ['WORKSPACE_NAMESPACE']\n",
+    "workspace = os.environ['WORKSPACE_NAME']\n",
+    "\n",
+    "# Getting list of tables in the data table \n",
+    "data_tables_list = fapi.get_entities_with_type(namespace, workspace).json()\n",
+    "\n",
+    "# Assigning column search key\n",
+    "column_name_search_key = \"object_id\"\n",
+    "\n",
+    "# Creating the master DRS urls dictionary \n",
+    "DRS_urls_dict = {}\n",
+    "\n",
+    "# Going through each table in the data table and pull out the contents of the object_id column\n",
+    "for table in data_tables_list:\n",
+    "    table_colums = table[\"attributes\"]\n",
+    "    object_id_column = dict(filter(lambda item: column_name_search_key in item[0], table_colums.items())) \n",
+    "    \n",
+    "    # Checking if the object_id column's contents is a string and start with 'drs://' \n",
+    "    if object_id_column:\n",
+    "        DRS_url = next(iter(object_id_column.values()))\n",
+    "        if type(DRS_url) is str and DRS_url[0:6] == 'drs://':\n",
```

Comment:
i think this will error if `DRS_url` is a string shorter than 6 characters... using `DRS_url.startswith('drs://')` should be safer